test/e2e/storage/regional_pd.go

```diff
@@ -264,6 +276,94 @@ func testZonalFailover(c clientset.Interface, ns string) {
 
 }
 
+func testRegionalDelayedBinding(c clientset.Interface, ns string) {
+	test := storageClassTest{
+		name:        "Regional PD storage class with waitForFirstConsumer test on GCE",
+		provisioner: "kubernetes.io/gce-pd",
+		parameters: map[string]string{
+			"type":             "pd-standard",
+			"replication-type": "regional-pd",
+		},
+		claimSize:    "2Gi",
+		delayBinding: true,
+	}
+
+	suffix := "delayed-regional"
+	class := newStorageClass(test, ns, suffix)
+	claim := newClaim(test, ns, suffix)
+	claim.Spec.StorageClassName = &class.Name
+	pv, node := testBindingWaitForFirstConsumer(c, claim, class)
+	if node == nil {
+		framework.Failf("unexpected nil node found")
+	}
+	zone, ok := node.Labels[kubeletapis.LabelZoneFailureDomain]
+	if !ok {
+		framework.Failf("label %s not found on Node", kubeletapis.LabelZoneFailureDomain)
+	}
+	checkZoneFromLabelAndAffinity(pv, zone, false)
+}
+
+func testRegionalAllowedTopologies(c clientset.Interface, ns string) {
+	test := storageClassTest{
+		name:        "Regional PD storage class with allowedTopologies test on GCE",
+		provisioner: "kubernetes.io/gce-pd",
+		parameters: map[string]string{
+			"type":             "pd-standard",
+			"replication-type": "regional-pd",
+		},
+		claimSize:    "2Gi",
+		expectedSize: "2Gi",
+	}
+
+	suffix := "topo-regional"
+	class := newStorageClass(test, ns, suffix)
+	zones := getTwoRandomZones(c)
+	addAllowedTopologiesToStorageClass(c, class, zones)
+	claim := newClaim(test, ns, suffix)
+	claim.Spec.StorageClassName = &class.Name
+	pv := testDynamicProvisioning(test, c, claim, class)
+	checkZonesFromLabelAndAffinity(pv, sets.NewString(zones...), true)
+}
+
+func testRegionalAllowedTopologiesWithDelayedBinding(c clientset.Interface, ns string) {
+	test := storageClassTest{
+		name:        "Regional PD storage class with allowedTopologies and waitForFirstConsumer test on GCE",
+		provisioner: "kubernetes.io/gce-pd",
+		parameters: map[string]string{
+			"type":             "pd-standard",
+			"replication-type": "regional-pd",
+		},
+		claimSize:    "2Gi",
+		delayBinding: true,
+	}
+
+	suffix := "topo-delayed-regional"
+	class := newStorageClass(test, ns, suffix)
+	topoZones := getTwoRandomZones(c)
+	addAllowedTopologiesToStorageClass(c, class, topoZones)
+	claim := newClaim(test, ns, suffix)
+	claim.Spec.StorageClassName = &class.Name
+	pv, node := testBindingWaitForFirstConsumer(c, claim, class)
+	if node == nil {
+		framework.Failf("unexpected nil node found")
+	}
+	nodeZone, ok := node.Labels[kubeletapis.LabelZoneFailureDomain]
+	if !ok {
+		framework.Failf("label %s not found on Node", kubeletapis.LabelZoneFailureDomain)
+	}
+	zoneFound := false
+	for _, zone := range topoZones {
+		if zone == nodeZone {
+			zoneFound = true
+			break
+		}
+	}
+	if !zoneFound {
+		framework.Failf("zones specified in AllowedTopologies: %v does not contain zone of node where PV got provisioned: %s", topoZones, nodeZone)
+	}
+	checkZoneFromLabelAndAffinity(pv, nodeZone, false)
```

Comment:
should this check that both zones were specified?

Reply:
Indeed. Thanks for catching this. I have tightened the check.